File: src/setup_handlebars.rb
require "ruby-handlebars"

def setupHandlebars
  hbs = Handlebars::Handlebars.new

  hbs.register_helper('isZero') do |context, value, block, else_block|
    if value == 0
      block.fn(context)
    elsif else_block
      else_block.fn(context)
    else
      ''
    end
  end


  templates_directory = File.join(File.dirname(__FILE__), '../templates/')
  compiled_templates = {}

  Dir.entries(templates_directory).each do |file|
    next unless file.end_with?('.hbs')

    file_path = File.join(templates_directory, file)
    template_name = File.basename(file, '.hbs')
    template_string = File.read(file_path)

    hbs.register_partial(template_name, template_string)

    compiled_templates[template_name] = hbs.compile(template_string)
  end

  return compiled_templates
end

File: src/setupHandlebars.js
import Handlebars from 'handlebars';
import fs from 'fs/promises';
import { getPath } from './utils.js';

const templatesDir = '/templates';

export default async function SetupHandlebars() {
  const templates = {};

  const templateFiles = await fs.readdir(getPath(templatesDir));
  for (const file of templateFiles) {
    if (file.endsWith('.hbs')) {
      const filePath = getPath(`${templatesDir}/${file}`);
      const templateString = await fs.readFile(filePath, 'utf8');
      Handlebars.registerPartial(file.split('.')[0], templateString);
      templates[file.split('.')[0]] = Handlebars.compile(templateString);
    }
  }

  Handlebars.registerHelper('isZero', function(value, options) {
    if (value === 0) {
      return options.fn(this);
    }
    else if (options.inverse) {
      return options.inverse(this);
    }
    return '';
  });

  return templates;
}
